### ISSUE TITLE
#4852 does not fix issues with methods 

### ISSUE THREAD

**Issue by sylee957** [microsoft/pyright#4914]
**Describe the bug**

#4852 is not applied same to methods.

**To Reproduce**

```python3
from collections.abc import Callable
from typing import TypeVar

T = TypeVar("T")
A = TypeVar("A")
B = TypeVar("B")
C = TypeVar("C")
X = TypeVar("X")
Y = TypeVar("Y")
Z = TypeVar("Z")


def identity(x: T) -> T:
    return x


def triple(
    f: Callable[[A], X], g: Callable[[B], Y], h: Callable[[C], Z]
) -> Callable[[A, B, C], tuple[X, Y, Z]]:
    def wrapped(a: A, b: B, c: C) -> tuple[X, Y, Z]:
        return f(a), g(b), h(c)

    return wrapped


class Foo:
    def identity(self, x: T) -> T:
        return x

    def test_1(self, f: Callable[[A], X]) -> Callable[[A, B, C], tuple[X, B, C]]:
        return triple(f, self.identity, self.identity)  # type check fails

    def test_2(self, f: Callable[[A], X]) -> Callable[[A, B, C], tuple[X, B, C]]:
        return triple(f, identity, identity)
```

**Expected behavior**

`test_1` should pass the type check

**Screenshots or Code**

**VS Code extension or command-line**

I use pyright 1.1.302

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. Yep, I didn't test this case, and it didn't work because of an optimization that I had added in when I implemented the support earlier. In the case of a method that is accessed through a `self` object, the method is partially specialized based on the value of `self`, and this partial specialization disabled the logic that detects conflicting signatures.

This will be fixed in the next release.
